### PR TITLE
refactor(chip): ripple inner element fixup

### PR DESCRIPTION
**Pull request by aprigogin**
Added remainder of the changes that were needed to move ripple element to be internal to the chip.

### CODE REVIEW

**Comment by aprigogin** [comment]
@googlebot I consent

File: packages/mdc-chips/_mixins.scss [hunk 1]
```diff
@@ -429,6 +448,18 @@ $mdc-chip-ripple-target: ".mdc-chip__ripple";
...
     border-width: $width;
   }
+
+  $horizontal-margin: 0 - $horizontal-padding-value - $width;
+  $vertical-margin: 0 - $width;
+
+  #{$mdc-chip-ripple-target} {
+    @include mdc-feature-targets($feat-structure) {
+      top: $vertical-margin;
+      margin-right: $horizontal-margin;
```
Comment (abhiomkar) [inline packages/mdc-chips/_mixins.scss]: We do not need negative margin & transparent border if ripple is absolutely positioned inside chip.

We would need to explicitly set top & left position to `0` to start from top-left corner relative to its container.

Comment (aprigogin) [reply packages/mdc-chips/_mixins.scss]: good point... you do need transparent borders, but do not need negative margins, my mistake.

File: packages/mdc-chips/_mixins.scss [hunk 2]
```diff
@@ -438,6 +465,15 @@ $mdc-chip-ripple-target: ".mdc-chip__ripple";
     padding-right: $padding;
     padding-left: $padding;
   }
+
+  $margin: 0 - $padding;
+
+  #{$mdc-chip-ripple-target} {
+    @include mdc-feature-targets($feat-structure) {
+      margin-right: $margin;
```
Comment (abhiomkar) [inline packages/mdc-chips/_mixins.scss]: Can we remove negative margins from here too?

File: packages/mdc-chips/_mixins.scss [hunk 3]
```diff
@@ -429,6 +448,14 @@ $mdc-chip-ripple-target: ".mdc-chip__ripple";
     padding-left: $horizontal-padding-value;
     border-width: $width;
   }
+
+  #{$mdc-chip-ripple-target} {
+    @include mdc-feature-targets($feat-structure) {
+      top: -$width;
```
Comment (abhiomkar) [inline packages/mdc-chips/_mixins.scss]: top & left can be set to `0` which is the same thing.

File: packages/mdc-chips/_mixins.scss [hunk 4]
```diff
@@ -438,6 +465,15 @@ $mdc-chip-ripple-target: ".mdc-chip__ripple";
     padding-right: $padding;
     padding-left: $padding;
   }
+
+  $margin: 0 - $padding;
+
```
Comment (joyzhong) [inline packages/mdc-chips/_mixins.scss]: nit: -1 newline

File: packages/mdc-chips/_mixins.scss (hunk 3 above)
Comment (joyzhong) [reply packages/mdc-chips/_mixins.scss]: I think this is necessary in order to compensate for the chip border. top/left of 0 will position the ripple element inside the border.

Comment (abhiomkar) [reply packages/mdc-chips/_mixins.scss]: Ah that's right. this is to offset transparent border-width added to ripple.

File: packages/mdc-chips/_mixins.scss (hunk 2 above)
Comment (aprigogin) [reply packages/mdc-chips/_mixins.scss]: Argh!!! had it removed from corresponding CL, but missed it on the PR.

File: packages/mdc-chips/_mixins.scss [hunk 5]
```diff
@@ -438,6 +465,15 @@ $mdc-chip-ripple-target: ".mdc-chip__ripple";
     padding-right: $padding;
     padding-left: $padding;
   }
+
+  $margin: 0 - $padding;
```
Comment (abhiomkar) [inline packages/mdc-chips/_mixins.scss]: nit: `$margin: -$padding;`

Comment (aprigogin) [reply packages/mdc-chips/_mixins.scss]: That whole chunk should have been removed

File: packages/mdc-chips/_mixins.scss (hunk 4 above)
Comment (aprigogin) [reply packages/mdc-chips/_mixins.scss]: This whole chunk should not have been there. Removed from CL, but forgot to update PR.

**Pull request merged**